src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -2532,29 +2538,49 @@ class ArnoldGlobals
 
 		void render()
 		{
-			updateCamera();
+
+
 			AiNodeSetInt(
 				AiUniverseGetOptions(), g_aaSeedArnoldString,
 				m_aaSeed.get_value_or( m_frame.get_value_or( 1 ) )
 			);
 
+
 			// Do the appropriate render based on
 			// m_renderType.
 			switch( m_renderType )
 			{
 				case IECoreScenePreview::Renderer::Batch :
 				{
-					const int result = AiRender( AI_RENDER_MODE_CAMERA );
-					if( result != AI_SUCCESS )
+					// Loop through all cameras referenced by any current outputs,
+					// and do a render for each
+					std::vector<std::string> cameraOverrides;
```

Comment:
This needs to be a set! Currently we're doing one render per output, even if they all have the same camera...